mozilla-release/mobile/android/app/src/photon/java/org/mozilla/gecko/toolbar/ToolbarDisplayLayout.java

```diff
@@ -197,6 +198,15 @@ public void onLayoutChange(View v, int left, int top, int right, int bottom, int
 
         mStop = (ThemedImageButton) findViewById(R.id.stop);
         mPageActionLayout = (PageActionLayout) findViewById(R.id.page_action_layout);
+        // update Title Bar width depends on PageActions visibility
+        mPageActionLayout.getViewTreeObserver().addOnGlobalLayoutListener(new ViewTreeObserver.OnGlobalLayoutListener() {
```

Comment:
Could not this be done by overriding `onLayout(...)`?

Reply:
No,
I want to be notified when the view is visible and when is hide,

https://developer.android.com/reference/android/view/View.html#onLayout(boolean,%20int,%20int,%20int,%20int)

https://developer.android.com/reference/android/view/ViewTreeObserver.html#addOnGlobalLayoutListener(android.view.ViewTreeObserver.OnGlobalLayoutListener)